scipy/optimize/_shgo.py

```diff
@@ -1446,7 +1446,10 @@ def fun_ref(self):
                         break  # Breaks the g loop
 
             if eval_f:
-                self.F[i] = self.func(self.C[i, :], *self.args)
+                # sometimes self.func returns a scalar, sometimes [x] or [[[[x]]]].
+                # TODO get some consistency in and remove the reshape
+                val = self.func(self.C[i, :], *self.args)
+                self.F[i] = np.asarray(val).reshape(1)[0]
```

Comment:
For this pattern, you'd be faster with
```suggestion
                self.F[i] = np.asarray(val).reshape(())
```

Reply:
There are a few other places in this patch where you could do the same thing